moveit_ros/moveit_servo/src/servo.cpp

```diff
@@ -459,36 +453,36 @@ KinematicState Servo::getNextJointState(const ServoInput& command)
     joint_position_delta *= collision_velocity_scale_;
 
     // Compute the next joint positions based on the joint position deltas
-    target_joint_positions = current_joint_positions + joint_position_delta;
+    target_state.positions = current_state.positions + joint_position_delta;
 
     // TODO : apply filtering to the velocity instead of position
     // Apply smoothing to the positions if a smoother was provided.
     // Update filter state and apply filtering in position domain
     if (smoother_)
     {
-      smoother_->reset(current_state.positions);
-      smoother_->doSmoothing(target_state.positions);
+      smoother_->reset(current_state.positions, current_state.velocities, current_state.accelerations);
```

Comment:
Not related to this PR, but does it even make sense to call `reset()` here? It seems like we should only do that after the arm has halted.

If you agree, let's open an issue

Reply:
Since the current state of the joints might be different than the  `target_state` passed to `doSmoothing` in previous iteration, wouldn't calling `reset` at each iteration  be useful to avoid discrepancies?

Reply:
Yeah, i don't think this is correct. Looking at the code for `reset()` it does this:

```
  previous_measurements_[0] = data;
  previous_measurements_[1] = data;
  previous_filtered_measurement_ = data;
```

So it completely wipes out all filter information. It's like there's no filtering at all. My bad for not catching this before.